pennylane/devices/experimental/device_api.py

```diff
@@ -185,7 +185,7 @@ def blank_postprocessing_fn(res):
             """
             return res
 
-        circuit_batch = (circuits,) if isinstance(circuits, QuantumTape) else circuits
+        circuit_batch = (circuits,) if isinstance(circuits, QuantumScript) else circuits
```

Comment:
`QuantumTape` is kept in the documentation because that is the name external power users expect.

`QuantumTape` is the name we must use in documentation, but it's not the object we need to use in implementation.

`QuantumTape` caries around a global threading lock and queue, and is intrinsically mutable. We don't want that in our circuit representation.

Reply:
Sor for all development purposes we should rely on `QuantumScript`, right? Thank you for clarifying this was super helpful.
Should we update [here](https://github.com/PennyLaneAI/pennylane/pull/4111/files#diff-441d185a784339b0f4d609e38f6f7aca7a03ee6f547f2bb1f22f2f937c3a418cR28-R29), too?

Reply:
I guess it depends on whether type hints are consider part of the documentation or not.